source/Engine.cpp

```diff
@@ -552,28 +552,29 @@ void Engine::Step(bool isActive)
 	if(nextZoom)
 	{
 		// TODO: std::exchange
-		zoom = nextZoom;
+		trueZoom = nextZoom;
 		nextZoom = 0.;
 	}
 	// Smoothly zoom in and out.
 	if(isActive)
 	{
 		double zoomTarget = Preferences::ViewZoom();
-		if(zoom != zoomTarget)
+		if(trueZoom != zoomTarget)
 		{
 			static const double ZOOM_SPEED = .05;
 
 			// Define zoom speed bounds to prevent asymptotic behavior.
 			static const double MAX_SPEED = .05;
 			static const double MIN_SPEED = .002;
 
-			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(zoom) - log2(zoomTarget)) * ZOOM_SPEED));
-			if(zoom < zoomTarget)
-				nextZoom = min(zoomTarget, zoom * (1. + zoomRatio));
-			else if(zoom > zoomTarget)
-				nextZoom = max(zoomTarget, zoom * (1. / (1. + zoomRatio)));
+			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(trueZoom) - log2(zoomTarget)) * ZOOM_SPEED));
```

Comment:
line too long
```suggestion
			double zoomRatio = max(MIN_SPEED, min(MAX_SPEED, abs(log2(trueZoom) - log2(zoomTarget))
				* ZOOM_SPEED));
```

Reply:
How long is too long?

Reply:
maximum 100

Reply:
> Try to keep lines to about 80 characters in length, and wrap anything over 120 characters.
The style guide says that wrapping is necessary over 120. Wrapping it now would damage the flow of the code.